### PR TITLE
Read db passwords from env file

### PR DESCRIPTION
**Pull request by Vayel**

### CODE REVIEW

**Comment by Vayel** [comment]
@lbarro how can this PR benefite to this repo?

**Comment by Vayel** [comment]
@lbarro @julien-bayle @julienbayle @GuillaumeGautierLA I can split this into multiple PRs.

**Comment by julien-bayle** [comment]
This is a good idea. 
My collegue @labarro will merge it soon as I'm going to be out of office for a few months.

Thx for this PR.

**Review by julien-bayle — APPROVED** [review]

**Review by julien-bayle — APPROVED** [review]

**Review by julien-bayle — DISMISSED** [review]

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,3 @@
-data/
-.env
-shellinstall
\ No newline at end of file
+data/*
```
Comment (julien-bayle) [inline .gitignore]: Adding a star is useless

File: base/pyenv.template [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+DB_AUTHENTIC_PASS = '$DB_AUTHENTIC_PASS'
+
+RABBITMQ_DEFAULT_USER = '$RABBITMQ_DEFAULT_USER'
+RABBITMQ_DEFAULT_PASS = '$RABBITMQ_DEFAULT_PASS'
+RABBITMQ_PORT = '$RABBITMQ_PORT'
+
+ERROR_MAIL_AUTHOR = '$ERROR_MAIL_AUTHOR'
+ERROR_MAIL_ADDR = '$ERROR_MAIL_ADDR'
+
+MAILCATCHER_SMTP_PORT = '$MAILCATCHER_SMTP_PORT'
```
Comment (julien-bayle) [inline base/pyenv.template]: Let's rename it SMTP_PORT

File: check-install.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+set -eu
+
+docker exec authentic ls "/var/lib/authentic2-multitenant/tenants"
+docker exec combo ls "/var/lib/combo/tenants"
+docker exec fargo ls "/var/lib/fargo/tenants"
+docker exec hobo ls "/var/lib/hobo/tenants"
+docker exec passerelle ls "/var/lib/passerelle/tenants"
+docker exec wcs ls "/var/lib/wcs"
+
+echo "Installation OK"
```
Comment (julien-bayle) [inline check-install.sh]: Use wc -l to check that directory is not empty as the message always say "OK" even if installation is not OK

File: combo/Dockerfile [hunk 4]
```diff
@@ -3,19 +3,19 @@ FROM julienbayle/publik:latest-base
...
-    && rm -rf /var/lib/apt/lists/*
+# python3-django and python3-cryptography backports needed for combo
+RUN apt-get update
+RUN apt-get -t stretch-backports install -y --no-install-recommends python3-django python3-cryptography
+RUN apt-get install -y --no-install-recommends postgresql-
+# See https://dev.entrouvert.org/issues/38997
+RUN apt-get install -y --no-install-recommends python3-xstatic-roboto-fontface=0.4.3.2-1
+RUN apt-get install -y --no-install-recommends combo
+RUN apt-get install -y --no-install-recommends publik-base-theme
+RUN rm -rf /var/lib/apt/lists/*
```
Comment (julien-bayle) [inline combo/Dockerfile]: If you split this RUN in multiple line then the image size will be larger. There is no point in splitting it

**Comment by Vayel** [comment]
I had some problems making it work so I asked Entrouvert and, apparently, Publik wasn't designed to work on multiple containers. All components need to be installed on the same machine (or container).

I'm trying to merge the containers while keeping the automation and the configuration with Docker and env vars.

File: wcs/wcs.settings.py [hunk 5]
```diff
@@ -40,11 +42,18 @@
...
+            'filename': '/var/log/wcs/django.log',
+            'formatter': 'simple'
+        },
     },
     'loggers': {
 	'':{
-            'handlers': ['console'],
-            'level': 'INFO',
+            'handlers': ['console', 'file'],
+            'level': 'DEBUG',
```
Comment (julien-bayle) [inline wcs/wcs.settings.py]: Perhaps we should add a variable for this.

File: passerelle/passerelle.settings.py [hunk 6]
```diff
@@ -40,11 +41,18 @@
...
+            'filename': '/var/log/passerelle/django.log',
+            'formatter': 'simple'
+        },
     },
     'loggers': {
 	'':{
-            'handlers': ['console'],
-            'level': 'INFO',
+            'handlers': ['console', 'file'],
+            'level': 'DEBUG',
```
Comment (julien-bayle) [inline passerelle/passerelle.settings.py]: Perhaps we should add a variable for this.

File: hobo/hobo.settings.py [hunk 7]
```diff
@@ -41,11 +46,18 @@
...
+            'filename': '/var/log/hobo/django.log',
+            'formatter': 'simple'
+        },
     },
     'loggers': {
 	'':{
-            'handlers': ['console'],
-            'level': 'INFO',
+            'handlers': ['console', 'file'],
+            'level': 'DEBUG',
```
Comment (julien-bayle) [inline hobo/hobo.settings.py]: Perhaps we should add a variable for this.

File: fargo/fargo.settings.py [hunk 8]
```diff
@@ -40,11 +41,18 @@
...
+            'filename': '/var/log/fargo/django.log',
+            'formatter': 'simple'
+        },
     },
     'loggers': {
 	'':{
-            'handlers': ['console'],
-            'level': 'INFO',
+            'handlers': ['console', 'file'],
+            'level': 'DEBUG',
```
Comment (julien-bayle) [inline fargo/fargo.settings.py]: Perhaps we should add a variable for this.

File: install-on-debian.sh [hunk 9]
```diff
@@ -72,18 +76,6 @@ echo "Add some convenients tools"
 
 apt-get install -y build-essential gettext vim git
 
-echo "Add publik.bash to bash env of publik user..."
```
Comment (julien-bayle) [inline install-on-debian.sh]: Why do you remove these aliases ?

File: proxy/docker-entrypoint.sh [hunk 10]
```diff
@@ -33,28 +52,28 @@ function generatecertificate() {
 	fi
 }
 
-generateconf demarches combo http
```
Comment (julien-bayle) [inline proxy/docker-entrypoint.sh:36]: Should use the subdomains variable

File: docs/deploy-prod.md [hunk 11]
```diff
@@ -0,0 +1,6 @@
+# Déploiement de production
+
+Suivre cette procédure pour déployer la version de **production** des conteneurs
+sur une machine **accessible depuis Internet**.
+
+TODO
```
Comment (julien-bayle) [inline docs/deploy-prod.md]: Exclude this commit from PR as the doc is not ready yet

File: docs/deploy-local.md [hunk 12]
```diff
@@ -0,0 +1,6 @@
+# Déploiement local
+
+Suivre cette procédure pour déployer les conteneurs sur une machine **non accessible
+depuis Internet**.
+
+TODO
```
Comment (julien-bayle) [inline docs/deploy-local.md]: Exclude this commit from PR as the doc is not ready yet

File: docs/deploy-dev.md [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+
+Des certificats Let's encrypt (Fournisseur de certificats HTTPS gratuits) seront automatiquement générés durant le processus d'installation. C'est pourquoi il est important que le serveur soit accessible depuis internet et que les enregistrement DNS ait été préalablement configurées avant de commencer.
+
+Vous pouvez vérifier votre configuration DNS en réalisant un ping sur une adresse.
+
+Exemple :
+
+```
+ping combo.monserveur.fr
+```
```
Comment (julien-bayle) [inline docs/deploy-dev.md]: Exclude this commit from PR as the doc is not ready yet

File: authentic/authentic.config.py [hunk 14]
```diff
@@ -84,11 +88,18 @@
...
+            'filename': '/var/log/authentic2-multitenant/django.log',
+            'formatter': 'simple'
+        },
     },
     'loggers': {
 	'':{
-            'handlers': ['console'],
-            'level': 'INFO',
+            'handlers': ['console', 'file'],
+            'level': 'DEBUG',
```
Comment (julien-bayle) [inline authentic/authentic.config.py]: Should use a variable like LOGGER_LEVEL

File: authentic/authentic.config.py [hunk 15]
```diff
@@ -84,11 +88,18 @@
             'class': 'logging.StreamHandler',
             'formatter': 'simple'
         },
+        'file': {
+            'level': 'DEBUG',
+            'class': 'logging.FileHandler',
+            'filters': ['require_debug_true'],
+            'filename': '/var/log/authentic2-multitenant/django.log',
```
Comment (julien-bayle) [inline authentic/authentic.config.py]: Logs are available in systemd... adding a file log in production must be an option as a log rotate mecanism should also be configured (already available in systemd)

File: combo/combo.settings.py [hunk 16]
```diff
@@ -40,11 +41,18 @@
...
+            'filename': '/var/log/combo/django.log',
+            'formatter': 'simple'
+        },
     },
     'loggers': {
 	'':{
-            'handlers': ['console'],
-            'level': 'INFO',
+            'handlers': ['console', 'file'],
+            'level': 'DEBUG',
```
Comment (julien-bayle) [inline combo/combo.settings.py]: File logging and log level should be in a variable

File: docs/README.md [hunk 17]
```diff
@@ -0,0 +1,93 @@
...
+(/usr/local/share/ca-certificates) et déclarer le proxy dans la configuration docker
+(https://docs.docker.com/v1.13/engine/admin/systemd/#http-proxy).
+Il est également possible d'utiliser "--insecure" pour sync-os afin de faire des appels curls sans 
+vérification des certificats soit la commande complète "sudo ./sync-os.sh --insecure"
+
+Pour la suite, se référer à :
+
+* `docs/deploy-local.md` pour une installation sur une machine **non accessible** depuis Internet
+* `docs/deploy-dev.md` pour une installation de développement sur une machine **accessible** depuis Internet
+* `docs/deploy-prod.md` pour une installation de production sur une machine **accessible** depuis Internet
```
Comment (julien-bayle) [inline docs/README.md]: Good "new" documentation. 
Anyway, can you move it in another PR as could be merged separately ?

**Review by julien-bayle — CHANGES_REQUESTED** [review]

**Comment by Vayel** [comment]
@julien-bayle we can have a call if you want (I sent you my phone number on LinkedIn).

File: combo/Dockerfile (hunk 4 above)
Comment (Vayel) [reply combo/Dockerfile]: Ok. Can you explain what `rm -rf /var/lib/apt/lists/*` is for?

File: install-on-debian.sh (hunk 9 above)
Comment (Vayel) [reply install-on-debian.sh]: I find it cleaner to avoid "polluting" the user's environment outside the project.

File: combo/Dockerfile (hunk 4 above)
Comment (julien-bayle) [reply combo/Dockerfile]: This line removes apt cache files in the current image layer making it smaller (See best pratice https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#run)

File: install-on-debian.sh (hunk 9 above)
Comment (julien-bayle) [reply install-on-debian.sh]: Why not. However the main documentation refers to all these aliases... so removing it also breaks all the documentation (README.md). I propose to keep it for now and create a PR later that will add .sh script for all aliases, update main doc and remove the bash aliases file.

File: authentic/authentic.config.py (hunk 15 above)
Comment (julien-bayle) [reply authentic/authentic.config.py]: I just pushed in this branch an initial implementation for this

File: install-on-debian.sh (hunk 9 above)
Comment (julien-bayle) [reply install-on-debian.sh]: I restored this part to merge this PR

**Review by julien-bayle — APPROVED** [review]

**Comment by julien-bayle** [comment]
Thx !

**Pull request merged**